container/Containerfile

```diff
@@ -0,0 +1,233 @@
+#       submodule in the ceph repo or something to be included with ceph packages?
+#       @tchaikov @torchiaf @adk3798 @ktdreyer?
+#       ref: https://github.com/ceph/ceph-container/pull/1821
+#       note: scikit-learn packages are not included in ubi8 in ceph-container
+RUN \
+    dnf install -y --setopt=install_weak_deps=False dnf-plugins-core && \
+    dnf copr enable -y tchaikov/python-scikit-learn && \
+    dnf copr enable -y tchaikov/python3-asyncssh
+
+# Update package mgr
+RUN dnf update -y --setopt=install_weak_deps=False
+
+# Define and install packages
+# TODO: review these packages
+RUN \
+    # General
+    PACKAGES="ca-certificates" && \
+    # Ceph
+    # TODO: I removed these packages that were included in ganesha packages:
+    #       gcc, lua-devel, luarocks
+    #    They are listed as BuildRequires in ceph.spec.in, so I assume they are unnecessary
+    #    @mattbenjamin please advise about whether these are necessary for runtime
+    PACKAGES="$PACKAGES \
+        ceph-common${CEPH_PACKAGE_VERSION} \
+        ceph-exporter${CEPH_PACKAGE_VERSION} \
+        ceph-grafana-dashboards${CEPH_PACKAGE_VERSION} \
+        ceph-immutable-object-cache${CEPH_PACKAGE_VERSION} \
+        ceph-mds${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-cephadm${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-dashboard${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-diskprediction-local${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-k8sevents${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-rook${CEPH_PACKAGE_VERSION} \
+        ceph-mgr${CEPH_PACKAGE_VERSION} \
+        ceph-mon${CEPH_PACKAGE_VERSION} \
+        ceph-osd${CEPH_PACKAGE_VERSION} \
+        ceph-radosgw${CEPH_PACKAGE_VERSION} \
+        ceph-volume${CEPH_PACKAGE_VERSION} \
+        cephfs-mirror${CEPH_PACKAGE_VERSION} \
+        cephfs-top${CEPH_PACKAGE_VERSION} \
+        kmod \
+        libradosstriper1${CEPH_PACKAGE_VERSION} \
+        rbd-mirror${CEPH_PACKAGE_VERSION} \
+        $(cat crimson-packages.txt) \
+        " && \
+    # Ceph "Recommends"
+    PACKAGES="$PACKAGES \
+        nvme-cli \
+        python3-saml \
+        smartmontools \
+        " && \
+    # NFS-Ganesha
+    PACKAGES="$PACKAGES \
+        dbus-daemon \
+        nfs-ganesha-ceph${GANESHA_PACKAGE_VERSION} \
+        nfs-ganesha-rados-grace${GANESHA_PACKAGE_VERSION} \
+        nfs-ganesha-rados-urls${GANESHA_PACKAGE_VERSION} \
+        nfs-ganesha-rgw${GANESHA_PACKAGE_VERSION} \
+        nfs-ganesha${GANESHA_PACKAGE_VERSION} \
+        rpcbind \
+        sssd-client \
+        " && \
+    # ISCSI
+    PACKAGES="$PACKAGES \
+        ceph-iscsi \
+        tcmu-runner \
+        python3-rtslib \
+        " && \
+    # Ceph-CSI
+    # TODO: coordinate with @Madhu-1 to have Ceph-CSI install these itself if unused by ceph
+    #       @adk3798 does cephadm use these?
+    PACKAGES="$PACKAGES \
+        attr \
+        ceph-fuse${CEPH_PACKAGE_VERSION} \
+        rbd-nbd${CEPH_PACKAGE_VERSION} \
+        " && \
```

Comment:
@Madhu-1 @adk3798 let me know if you have thoughts around this. It seems to me like keeping the ceph-fuse and rbd-nbd drivers in the ceph image makes sense, even for non-CSI images. But if you have differing thoughts, let's discuss.

Also, @Madhu-1 is `attr` still needed, and if so, do you think it's broadly needed or only for Ceph-CSI?

Reply:
Cephadm doesn't use these directly at least. Can't say with certainty for any of the daemons cephadm deploys that use the ceph container image though.

Reply:
Leaving these for now.